walker.go

```diff
@@ -0,0 +1,57 @@
+package chi
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+type WalkFn func(method string, route string, handler http.Handler, middlewares ...func(http.Handler) http.Handler) error
+
+func Walk(r Routes, fn WalkFn) error {
```

Comment:
Example - print all routes in the following simple format:
```
POST /users
GET /users
GET /users/:id
PUT /users/:id
GET /users/:id/books
```

With `Walk()` fn, it's as simple as:
```go
chi.Walk(r, func(method string, route string, handler http.Handler) error {
	fmt.Printf("%v %v\n", method, route)
})
```